src/Parser.php

```diff
@@ -3175,9 +3176,16 @@ private function parseMemberAccessExpression($expression):MemberAccessExpression
     private function parseScopedPropertyAccessExpression($expression, $fallbackParentNode): ScopedPropertyAccessExpression {
         $scopedPropertyAccessExpression = new ScopedPropertyAccessExpression();
         $scopedPropertyAccessExpression->parent = $expression->parent ?? $fallbackParentNode;
+
         if ($expression instanceof Node) {
             $expression->parent = $scopedPropertyAccessExpression;
-            $scopedPropertyAccessExpression->scopeResolutionQualifier = $expression; // TODO ensure always a Node
+
+            // scopeResolutionQualifier does not accept `Node` but
```

Comment:
In followup PRs, that should hopefully be addressed by making parameter types more precise as well, assuming that phpstan can narrow a mix of subclasses of Node and definite non-subclassess to just the expected subclasses `Expression|QualifiedName` (haven't checked)